goal_src/engine/gfx/depth-cue.gc

```diff
@@ -5,3 +5,440 @@
 ;; name in dgo: depth-cue
 ;; dgos: GAME, ENGINE
 
+;; DECOMP BEGINS
+
+(define
+  *depth-cue-work*
+  (new 'static 'depth-cue-work
+    :texture-strip-tmpl
+    (new 'static 'dma-gif-packet
+      :dma-vif
+      (new 'static 'dma-packet
+        :dma
+        (new 'static 'dma-tag :qwc #x6 :id (dma-tag-id cnt))
+        :vif1
+        (new 'static 'vif-tag :imm #x6 :cmd (vif-cmd direct) :msk #x1)
+        )
+      :gif
+      (new 'static 'array uint64 2 #x50ab400000008001 #x43431)
+      )
+    :temp-strip-tmpl
+    (new 'static 'dma-gif-packet
+      :dma-vif
+      (new 'static 'dma-packet
+        :dma
+        (new 'static 'dma-tag :qwc #x6 :id (dma-tag-id cnt))
+        :vif1
+        (new 'static 'vif-tag :imm #x6 :cmd (vif-cmd direct) :msk #x1)
+        )
+      :gif
+      (new 'static 'array uint64 2 #x508b400000008001 #x43431)
+      )
+    :stencil-tmpl
+    (new 'static 'dma-gif-packet
+      :dma-vif
+      (new 'static 'dma-packet
+        :dma
+        (new 'static 'dma-tag :qwc #x17 :id (dma-tag-id cnt))
+        :vif1
+        (new 'static 'vif-tag :imm #x17 :cmd (vif-cmd direct) :msk #x1)
+        )
+      :gif
+      (new 'static 'array uint64 2 #xb003400000008002 #x44444444441)
+      )
+    :set-color
+    (new 'static 'vector4w :x #x80 :y #x80 :z #x80 :w #x80)
+    :draw-color
+    (new 'static 'vector4w :x #x84 :y #x84 :z #x84 :w #x80)
+    :depth
+    (new 'static 'depth-cue-data :data (new 'static 'vector :x 1.0 :z 163840.0))
+    :front
+    (new 'static 'depth-cue-data :data (new 'static 'vector :x 0.999 :y 0.4 :z 4096.0 :w 1.0))
+    )
+  )
+
+(defun depth-cue-set-stencil ((arg0 dma-buffer) (arg1 int) (arg2 int) (arg3 int) (arg4 dma-gif-packet))
+  (let* ((v1-0 arg0)
+         (t1-0 (the-as object (-> v1-0 base)))
+         )
+    (set! (-> (the-as vector4w t1-0) dword 0) (the-as uint #x10000004))
+    (set! (-> (the-as vector4w t1-0) z) 0)
+    (set! (-> (the-as vector4w t1-0) w) #x50000004)
+    (set! (-> v1-0 base) (&+ (the-as pointer t1-0) 16))
+    )
+  (let* ((v1-1 arg0)
+         (t1-2 (the-as object (-> v1-1 base)))
+         )
+    (set! (-> (the-as vector4w t1-2) dword 0) (make-u128 0 (the-as uint #x3000000000008001)))
+    (set! (-> (the-as vector4w t1-2) dword 1) (the-as uint #xeeeeeeeeeeeeeeee))
```

Comment:
This is a `gs-gif-tag` here. This "eeeeee" spam is the GIF register ID `a+d` over and over, a clear sign that this DMA packet is for setting the values of GS registers directly. You can find one in `set-display-gs-state` and see the type of casts I used there.